refactored a little bit of the code

dmitriy-p-petrov opened the pull request:
Hi! I looked at another ticket. I am leaving this here just so you can check it out and tell me if you don't like something while i'm still in the very beginning. 

What's done so far is:
- I added a couple more routes that do nothing for now;
- I changed `populateTemplate()` and all the helper functions it uses. I wanted to reuse them to populate the graphs for the other routes ( still not sure what to do with `makeColumnNames()` ).

And that's pretty much it. Everything else is just rearrangements.

After i figure out what to do with `makeColumnNames()` i'm gonna work on the `collector()` function. I'm just gonna add a few more checks for the other events. I'm not sure what to do with the `route-metrics.log` path. Maybe just receive it as a command-line argument?

bmacnaughton commented on servers/express.js:
```diff
@@ -2,32 +2,39 @@
 const app = express();
 
-const pathToLogFile = path.join(__dirname, '..', '..', 'node-route-metrics', 'route-metrics.log');
+const pathToLogFile = (process.argv.length > 2) ? process.argv[2] : 'router-metrics.log';
```
route-metrics, not router-metrics

bmacnaughton commented on servers/express.js:
```diff
@@ -2,32 +2,39 @@
-
-  const html = populateTemplate(htmlTemplate, chartOpt, columns, datarows);
+  
+  const html = populateTemplate(htmlTemplate, chartOpt, makeEventloopColumnNames(), datarows);
```
probably want to make event loop column names the same form as data rows, e.g.,

`const columnNames = makeEventloopColumnNames();` and then use that in the call to `populateTemplate()`.

bmacnaughton commented on servers/express.js:
```diff
@@ -2,32 +2,39 @@
+  const chartOpt = makeChartOptions({title: 'Insert title here', subtitle: '@contrast/route-metrics'});
+  const datarows = makeDataRows(memoryDataRows);
+  
+  const html = populateTemplate(htmlTemplate, chartOpt, makeMemoryColumnNames(), datarows);
```
ditto

bmacnaughton commented on servers/express.js:
```diff
@@ -2,32 +2,39 @@
-app.get('/memory', function(req, res) {
-  res.send('Work in progress');
+app.get('/cpu', function(req, res) {
+  const chartOpt = makeChartOptions({title: 'Time spent in user and system code respectively (ms)', subtitle: '@contrast/route-metrics'});
```
it's always good to have the units in the chart - nice.

bmacnaughton commented on servers/express.js:
```diff
@@ -65,51 +84,51 @@ const eventloopDataRows = [];
+        // Cpu data is in microseconds. Make it ms
+        cpuDataRows.push(`[${delta}, ${entry['cpuUserAvg'] / 1e3}, ${entry['cpuSystemAvg'] / 1e3}]`);
+        memoryDataRows.push(`[${delta}, ${rss}, ${heapTotal}, ${heapUsedAvg}, ${externalAvg}]`);
+      } else continue;
```
this i definitely don't like - to me `else` should always follow the form of the previous if/else-if blocks for consistency, so

```` 
} else {
     continue;
}
```

dmitriy-p-petrov commented:
Thanks for the feedback! I addressed all of your points.

About the `undefined` thing - i know it's `undefined` by default and it doesn't really matter, since it's used on the next line. I just like to explicitly define them even when they are empty.

I believe the arguments problem is fixed now. You can pass `-l` (it's an L) or `--logfile` for the logfile and `-p` or `--protocol` for the protocols. You can either pass one, two or neither of them. (i mean, you do kinda need to pass at least -p currently, because both servers use the same port by default, but yeah...)

I'm gonna finish the last chart now. I wanted to ask if all of them represent the same thing on the x-axis and what it is exactly? Looks like they are functions with respect to the number of entries in the logfile, but it said `seconds` down there, so i got kinda confused.
I'm also not sure if i should leave the `makeColumnNames()` functions so static?

bmacnaughton commented:
> I'm also not sure if i should leave the makeColumnNames() functions so static?

i don't understand this question.

>I'm gonna finish the last chart now. I wanted to ask if all of them represent the same thing on the x-axis and what it is exactly? Looks like they are functions with respect to the number of entries in the logfile, but it says seconds down there, so i got kinda confused.

yes, the x-axis is always time. that's why we're subtracting the base timestamp from each log entry's timestamp (then divide by 1000, because the times are ms). it might be nice to note in the x-axis title the base time, so something like:

`seconds since 2022-06-09T15:05:11.277Z`

bmacnaughton commented:
> I believe the arguments problem is fixed now. You can pass -l (it's an L) or --logfile for the logfile and -p or --protocol for the protocols. You can either pass one, two or neither of them. (i mean, you do kinda need to pass at least -p currently, because both servers use the same port by default, but yeah...)

thinking about this, it should probably take the protocol argument on the command line (not an option) because it's required. that way it will continue to work as it is but the logfile is now an optional argument.

dmitriy-p-petrov commented:
Hello!

I'm once again asking for more details.
Looked at the ticket again. I think the first few points are done to some extend.

- Not sure what "configurable port" means, though... isn't it already configurable, since it's not really optional currently?
- I also don't understand timed-fetch part
